fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -3324,43 +3325,202 @@ public ParseNode visitHelpStatement(StarRocksParser.HelpStatementContext context
     }
 
     // ------------------------------------------- Backup Store Statement ----------------------------------------------
-    @Override
-    public ParseNode visitBackupStatement(StarRocksParser.BackupStatementContext context) {
-        QualifiedName qualifiedName = getQualifiedName(context.qualifiedName());
-        LabelName labelName = qualifiedNameToLabelName(qualifiedName);
+    private ParseNode getObjectRef(StarRocksParser.QualifiedNameContext qualifiedNameContext,
+                                   StarRocksParser.PartitionNamesContext partitionNamesContext,
+                                   boolean isFunction, String alias, NodePosition position) {
+        if (isFunction) {
+            String functionName = getQualifiedName(qualifiedNameContext).toString();
+            FunctionName fnName = FunctionName.createFnName(functionName);
+            FunctionRef fnRef = new FunctionRef(fnName, alias, position);
+
+            return fnRef;
+        } else {
+            TableName tableName = qualifiedNameToTableName(getQualifiedName(qualifiedNameContext));
+            PartitionNames partitionNames = null;
+            if (partitionNamesContext != null) {
+                partitionNames = (PartitionNames) visit(partitionNamesContext);
+            }
+            TableRef tableRef = new TableRef(tableName, alias, partitionNames, position);
+            return tableRef;
+        }
+    }
+
+    private ParseNode parseBackupRestoreStatement(ParserRuleContext context) {
+        StarRocksParser.BackupStatementContext backupContext = null;
+        StarRocksParser.RestoreStatementContext restoreContext = null;
+
+        if (context instanceof StarRocksParser.RestoreStatementContext) {
+            restoreContext = (StarRocksParser.RestoreStatementContext) context;
+        } else {
+            backupContext = (StarRocksParser.BackupStatementContext) context;
+        }
+
+        boolean specifyDbExplicitly =
+                            backupContext != null ? (backupContext.DATABASE() != null) : (restoreContext.DATABASE() != null);
+
+        LabelName labelName = null;
+        String repoName = null;
+        // db which the snapshot should be restored in
+        String dbAlias = null;
+        // db name in snapshot meta data
+        String originDb = null;
+
+        boolean withOnClause = false;
+        boolean allTable = false;
```

Comment:
I think, it will be simpler to use vector to store allXxx and Refs, like:
1. ArrayList<Boolean> allObjects = false * 4.  with enum TABLE/MV/VIEW/FUNCTION.
2. ArrayList<ArrayList<>> allRefs = {mixTblRefs, mixTblRefs, mixTblRefs, new List<FunctionRef>()};

Reply:
Fixed NO.1, for NO.2 it seems that keep the original implementation is better, because we will treat `TABLE`/`MV`/`VIEW` as `TableRef`.